xblock/mixins.py

```diff
@@ -610,3 +611,43 @@ def has_support(self, view, functionality):
             True or False
         """
         return hasattr(view, "_supports") and functionality in view._supports  # pylint: disable=protected-access
+
+
+class CompletableXBlockMixin(object):
+    """
+    This mixin sets attributes and provides helper method to integrate XBlock with Completion API.
+    """
+
+    has_custom_completion = True
+    completion_method = 'completable'
```

Comment:
I mentioned on the ticket that I got a request from edx to change completion method to an enum or a set of constants.  Would you mind adding that to this PR?  Needed values:

* completable
* aggregator
* excluded

Reply:
@jcdyer I think I've lost track of this request - will do tomorrow.